userspace/falco/app_actions/gain_lock.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "application.h"
+#include <sys/file.h>
+
+using namespace falco::app;
+
+#define FALCO_LOCK_FILE "falco.lock"
```

Comment:
Note: this is in the current working directory, otherwise build would fail after linking (or, we should always build as root).
It is not great though, because one can easily escape the check by changing cwd.
This is something that needs to be addressed.